src/cratedb_fivetran_destination/schema_migration_helper.py

```diff
@@ -0,0 +1,356 @@
+                LOG_INFO,
+                f"[Migrate:RenameTable] from={rt.from_table} to={rt.to_table} schema={schema}",
+            )
+            return destination_sdk_pb2.MigrateResponse(success=True)
+
+        if entity_case == "rename_column":
+            rename_column = rename_op.rename_column
+            sql = f'ALTER TABLE "{schema}"."{table}" RENAME "{rename_column.from_column}" TO "{rename_column.to_column}";'
+            with self.engine.connect() as conn:
+                conn.execute(sa.text(sql))
+
+            log_message(
+                LOG_INFO,
+                f"[Migrate:RenameColumn] table={schema}.{table} from_col={rename_column.from_column} to_col={rename_column.to_column}",
+            )
+            return destination_sdk_pb2.MigrateResponse(success=True)
+
+        log_message(LOG_WARNING, "[Migrate:Rename] No rename entity specified")
+        return destination_sdk_pb2.MigrateResponse(unsupported=True)
+
+    def handle_add(self, add_op, schema, table, table_obj: common_pb2.Table):
+        """Handles add operations (add column in history mode, add column with default value)."""
+        entity_case = add_op.WhichOneof("entity")
+
+        if entity_case == "add_column_in_history_mode":
+            # table-map manipulation to simulate add column in history mode, replace with actual logic.
+            add_col_history_mode = add_op.add_column_in_history_mode
+            table_obj = self.table_map.get(table)
+            if table_obj:
+                new_col = table_obj.columns.add()
+                new_col.name = add_col_history_mode.column
+                new_col.type = add_col_history_mode.column_type
+
+            log_message(
+                LOG_INFO,
+                f"[Migrate:AddColumnHistory] table={schema}.{table} column={add_col_history_mode.column} type={add_col_history_mode.column_type} default={add_col_history_mode.default_value} op_ts={add_col_history_mode.operation_timestamp}",
+            )
+            return destination_sdk_pb2.MigrateResponse(success=True)
+
+        if entity_case == "add_column_with_default_value":
+            add_col_default_with_value = add_op.add_column_with_default_value
+
+            new_col = table_obj.columns.add()
+            new_col.name = add_col_default_with_value.column
+            new_col.type = add_col_default_with_value.column_type
+            type_ = TypeMap.to_cratedb(new_col.type, new_col.params)
+            sql = f'ALTER TABLE "{schema}"."{table}" ADD COLUMN "{new_col.name}" {type_};'
+            with self.engine.connect() as conn:
+                conn.execute(sa.text(sql))
+
+            log_message(
+                LOG_INFO,
+                f"[Migrate:AddColumnDefault] table={schema}.{table} column={add_col_default_with_value.column} type={add_col_default_with_value.column_type} default={add_col_default_with_value.default_value}",
+            )
+            return destination_sdk_pb2.MigrateResponse(success=True)
```

Comment:
CrateDB's `ALTER TABLE ... ADD COLUMN ... DEFAULT` statement needs more love.

- https://github.com/crate/cratedb-fivetran-destination/issues/111
- https://github.com/crate/crate/issues/18783

Reply:
The behaviour of the operation will not change with CrateDB, so the current implementation here is valid.